File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
import java.util.*;

class Solution {
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        
        // 1. 시간 초단위로 변환
        int videoLength = timeToSeconds(video_len);
        int position = timeToSeconds(pos);
        int opStart = timeToSeconds(op_start);
        int opEnd = timeToSeconds(op_end);

        // 2. 사용자 command 수행
        for (String command : commands) {
            // 오프닝 구간 확인
            if (position >= opStart && position <= opEnd) {
                position = opEnd;
            }
            
            // prev
            if (command.equals("prev")) {
                position = Math.max(0, position - 10);

            }
            // next
            else if (command.equals("next")) {
                position = Math.min(videoLength, position + 10);
            }
            
            // 오프닝 구간 확인
            if (position >= opStart && position <= opEnd) {
                position = opEnd;
            }
        }
        
        return secondsToTime(position);
    }

    // 시간 > 초 변환
    private int timeToSeconds(String time) {
        String[] parts = time.split(":");
        int minutes = Integer.parseInt(parts[0]);
        int seconds = Integer.parseInt(parts[1]);
        return minutes * 60 + seconds;
    }

    // 초 > 시간
    private String secondsToTime(int totalSeconds) {
        int minutes = totalSeconds / 60;
        int seconds = totalSeconds % 60;
        return String.format("%02d:%02d", minutes, seconds);
    }
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def solution(video_len, pos, op_start, op_end, commands):
    
    # 1. 시간 초단위로 전부 변환하기
    video_mm, video_ss = map(int, video_len.split(':'))
    pos_mm, pos_ss = map(int, pos.split(':'))
    op_start_mm, op_start_ss = map(int, op_start.split(':'))
    op_end_mm, op_end_ss = map(int, op_end.split(':'))
    
    video_len = video_mm*60 + video_ss
    pos = pos_mm*60 + pos_ss
    op_start = op_start_mm*60 + op_start_ss
    op_end = op_end_mm*60 + op_end_ss
    
    # 명령어 수행 전에 현재 위치가 오프닝 구간이면 오프닝 마지막으로 이동하기
    if op_start <= pos < op_end:
            pos = op_end
            
    # 2. 명령어 수행하기
    for command in commands:
        if command == 'prev':
            pos -= 10
            
            # 만약 영상 시작 전보다 이전이면 영상 시작점으로 이동
            if pos < 0:
                pos = 0
        else:
            pos += 10
            
            # 영상 끝을 넘어가면 영상 끝 시점으로 이동
            if pos > video_len:
                pos = video_len
            
        # 오프닝 구간 확인
        if op_start <= pos < op_end:
            pos = op_end
            
    
    # 3. 현재 시간 분,초로 다시 변환
    pos_mm = pos // 60
    pos_ss = pos % 60
        
    answer = str(pos_mm).rjust(2, '0')+ ':' + str(pos_ss).rjust(2, '0')
    return answer